src/ray/raylet/node_manager.cc

```diff
@@ -844,23 +844,20 @@ void NodeManager::ProcessClientMessage(
 void NodeManager::ProcessRegisterClientRequestMessage(
     const std::shared_ptr<LocalClientConnection> &client, const uint8_t *message_data) {
   auto message = flatbuffers::GetRoot<protocol::RegisterClientRequest>(message_data);
-  client->SetClientID(from_flatbuf<ClientID>(*message->client_id()));
+  client->SetClientID(from_flatbuf<ClientID>(*message->worker_id()));
   auto worker =
       std::make_shared<Worker>(message->worker_pid(), message->language(), client);
   if (message->is_worker()) {
     // Register the new worker.
     worker_pool_.RegisterWorker(std::move(worker));
     DispatchTasks(local_queues_.GetReadyTasksWithResources());
   } else {
-    // Register the new driver. Note that here the driver_id in RegisterClientRequest
-    // message is actually the ID of the driver task, while client_id represents the
-    // real driver ID, which can associate all the tasks/actors for a given driver,
-    // which is set to the worker ID.
-    // TODO(qwang): Use driver_task_id instead here.
-    const WorkerID driver_id = from_flatbuf<WorkerID>(*message->driver_id());
-    TaskID driver_task_id = TaskID::GetDriverTaskID(driver_id);
+    // Register the new driver.
+    const WorkerID driver_id = from_flatbuf<WorkerID>(*message->worker_id());
+    // Compute a dummy driver task id from a given driver.
+    const TaskID driver_task_id = TaskID::ComputeDriverTaskId(driver_id);
```

Comment:
This PR changes the behavior of these IDs. It uses the worker_id to compute driver_task_id (before it was *message->driver_id()) and *message->driver_id() for AssignJobId, before it was the worker_id.

Reply:
Sorry that this change led to your misunderstanding, but the behavior of these IDs was changed by me intentionally.

After this PR, the `worker_id` field indicates the id of the worker or the driver, and `job_id` field indicates the id of the job if the client is a driver(https://github.com/ray-project/ray/blob/master/src/ray/raylet/format/node_manager.fbs#L134toL138).

So it does make sense to compute a `root_driver_task_id` from the given driver id. And because the type of a driver id is `WorkerID`, it also make sense to declare the type of the param `driver_id` in `ComputeDriverTaskId` as `WorkerID` .

Reply:
I don't think we can just make this change without changing other parts of the code, like https://github.com/ray-project/ray/blob/master/python/ray/worker.py#L1727. I ran into this problem while working on https://github.com/ray-project/ray/pull/5076.

Reply:
Could you tell more details about the issue you ran into?

Reply:
as you mentioned, actually [this line](https://github.com/ray-project/ray/blob/master/python/ray/worker.py#L1727) should be changed to
 `worker.worker_id = compute_driver_id_from_job_id(job_id)`

 and currently, the code of `compute_driver_id_from_job_id` should be:
```python
def compute_driver_id_from_job_id(job_id):
    assert is_instance(job_id, JobID)
    return WorkerID(job_id.binary())
```

Does this make sense to you?

Reply:
I don't think that makes sense (the change you propose doesn't make a difference).

Reply:
The problem here is https://github.com/ray-project/ray/blob/master/python/ray/worker.py#L1863 I think. It would probably be best to just remove the job_id (as it is named in the current master) field from `RegisterClientRequest`.